lib/h3core.js

```diff
@@ -241,69 +270,75 @@ function coordinatesToGeoPolygon(coordinates, isGeoJson) {
  */
-function readLong(invocation) {
-    // Upper 32-bits of the long set via side-effect
-    const upper = C.getTempRet0();
-    return [invocation, upper];
+function readH3IndexFromPointer(cAddress, offset = 0) {
+    const lower = C.getValue(cAddress + SZ_H3INDEX * offset, 'i32');
+    const upper = C.getValue(cAddress + SZ_H3INDEX * offset + SZ_INT, 'i32');
+    // The lower bits are allowed to be 0s, but if the upper bits are 0
+    // this represents an invalid H3 index
+    return upper ? splitLongToH3Index(lower, upper) : null;
 }
 
 /**
- * Read an H3 index from a C return value. As with readLong, the argument to this function
- * is intended to be an invocation, e.g. readH3Index(H3.getSomeAddress()), to help ensure that
- * the temp value storing the upper bits of the long is still set.
+ * Read a boolean (32 bit) from a pointer to C memory.
  * @private
- * @param  {number} invocation  Invoked function returning a single H3 index
- * @return {H3Index}            H3 index, or null if index was invalid
+ * @param  {number} cAddress  Pointer to allocated C memory
+ * @param {number} offset     Offset, in number of booleans, in case we're
+ *                            reading an array
+ * @return {Boolean} Boolean value
  */
-function readH3Index(invocation) {
-    const [lower, upper] = readLong(invocation);
-    // The lower bits are allowed to be 0s, but if the upper bits are 0
-    // this represents an invalid H3 index
-    return upper ? splitLongToh3Index(lower, upper) : null;
+function readBooleanFromPointer(cAddress, offset = 0) {
+    const val = C.getValue(cAddress + SZ_INT * offset, 'i32');
+    return Boolean(val);
 }
 
 /**
- * Read an H3 index from a pointer to C memory.
+ * Read a double from a pointer to C memory.
  * @private
  * @param  {number} cAddress  Pointer to allocated C memory
- * @param {number} offset     Offset, in number of H3 indexes, in case we're
+ * @param {number} offset     Offset, in number of doubles, in case we're
  *                            reading an array
- * @return {H3Index}          H3 index, or null if index was invalid
+ * @return {number} Double value
  */
-function readH3IndexFromPointer(cAddress, offset = 0) {
-    const lower = C.getValue(cAddress + SZ_INT * offset * 2, 'i32');
-    const upper = C.getValue(cAddress + SZ_INT * (offset * 2 + 1), 'i32');
-    // The lower bits are allowed to be 0s, but if the upper bits are 0
-    // this represents an invalid H3 index
-    return upper ? splitLongToh3Index(lower, upper) : null;
+function readDoubleFromPointer(cAddress, offset = 0) {
+    return C.getValue(cAddress + SZ_DBL * offset, 'double');
+}
+
+/**
+ * Read a 64-bit int from a pointer to C memory into a JS 64-bit float.
+ * Note that this may lose precision if larger than MAX_SAFE_INTEGER
+ * @private
+ * @param  {number} cAddress  Pointer to allocated C memory
+ * @return {number} Double value
+ */
+function readInt64FromPointer(cAddress) {
```

Comment:
Seems like the function name should indicate a cast is happening. That kind of precision losing cast can be an unexpected operation.

Reply:
I guess I can update to `readInt64AsDoubleFromPointer` - a little wordy, but accurate

Reply:
It's worth noting though that losing precision will be very rare - the total count of res 15 tiles is still lower than `MAX_SAFE_INTEGER`. I can't think of a scenario in our current usage where we would deal with an `int64` value that's bigger than this.